### PR TITLE
Change Uyuni testing docker images to Leap 15.2

### PR DESCRIPTION
**Pull request by juliogonzalez**
## What does this PR change?

Change Uyuni testing docker images to Leap 15.2

I built the images locally. Didn't see any errors.

## GUI diff

No difference. Testing stuff.

- [x] **DONE**

## Documentation
- No documentation needed: Testing stuff.

- [x] **DONE**

## Test coverage
- No tests: We are actually changing tests

- [x] **DONE**

## Links

Related: https://github.com/SUSE/spacewalk/issues/11749

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: susemanager-utils/testing/docker/master/uyuni-push-to-obs/add_repositories.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
 set -e
 
-zypper ar -f https://download.opensuse.org/repositories/home:/mcalmer:/tito/openSUSE_Leap_15.1/home:mcalmer:tito.repo
-zypper ar -f http://download.suse.de/ibs/SUSE:/CA/openSUSE_Leap_15.1/SUSE:CA.repo
+zypper ar -f https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Utils/openSUSE_Leap_15.2/systemsmanagement:Uyuni:Utils.repo
```
Comment (juliogonzalez) [inline susemanager-utils/testing/docker/master/uyuni-push-to-obs/add_repositories.sh:4]: No need to update mcalmer's project anymore. Remember we moved tito to Uyuni.

**Comment by juliogonzalez** [comment]
Now that I reviewed the changes more, I wonder what happens with `scripts/schema_migration_test_pgsql-31to41.sh`

I am reviewing if we shouldn't change it as well.

**Comment by juliogonzalez** [comment]
In fact it's funny, that script is called 31to41 but it contains:
```
rpm -ivh /root/susemanager-schema-4.0.1-1.2.noarch.rpm
```
And then it does:
```
# THE NEXT BLOCK CAN BE COMMENTED OUT WHEN WE HAVE THE LINK PACKAGED
for v in `seq 30 -1 1`; do
    minusone=$(($v-1))
    if [ -d /etc/sysconfig/rhn/schema-upgrade/susemanager-schema-3.2.$minusone-to-susemanager-schema-3.2.$v ]; then
        if [ ! -d /etc/sysconfig/rhn/schema-upgrade/susemanager-schema-3.2.$v-to-susemanager-schema-4.0.0 ]; then
            mkdir /etc/sysconfig/rhn/schema-upgrade/susemanager-schema-3.2.$v-to-susemanager-schema-4.0.0
            # set hard this destination
            #export SUMA_TEST_SCHEMA_VERSION="4.0.1"
        fi
        break
    fi
done
```
And it also calls postgres10 (I guess it needs to change to postgres11)

**Comment by mcalmer** [comment]
@juliogonzalez you should change this. 

  https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/testing/docker/master/Makefile#L1

It is more important than the version in the Dockerfiles.

**Comment by mcalmer** [comment]
You need to check for a new schema RPM which is official and near to 4.1.<released> . It does not make sense to apply 20 years of schema upgrades.

**Comment by juliogonzalez** [comment]
@mcalmer, ACK, thanks. But it's 4.1.1 so: what now?

4.1.2? It's not strictly speaking 4.2.0 yet as we didn't bump the packages for now.

**Comment by juliogonzalez** [comment]
Before merging, I want to run the tests I changed, just in case. But for that I need obs-to-maven fixed first :-)

File: schema/spacewalk/upgrade/susemanager-schema-4.1.10-to-susemanager-schema-4.1.11/001-fix_suseChannelUserRoleView.sql [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+--
+-- This software is licensed to you under the GNU General Public License,
+-- version 2 (GPLv2). There is NO WARRANTY for this software, express or
+-- implied, including the implied warranties of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+-- along with this software; if not, see
+-- http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+--
+
+DROP VIEW IF EXISTS suseChannelUserRoleView;
```
Comment (juliogonzalez) [inline schema/spacewalk/upgrade/susemanager-schema-4.1.10-to-susemanager-schema-4.1.11/001-fix_suseChannelUserRoleView.sql:11]: This comes from the original script, but I wonder... do we need it if we have the `CREATE OR REPLACE` below :-?

File: schema/spacewalk/upgrade/susemanager-schema-4.1.10-to-susemanager-schema-4.1.11/001-fix_suseChannelUserRoleView.sql [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      WHEN combination.role = 'manage' AND combination.channel_org_id IS NULL OR combination.channel_org_id <> combination.user_org_id
+        THEN 'channel_not_owned'
+      -- otherwise, if channel is in a family without permissions for the user's org, then user can't subscribe it
+      WHEN combination.role = 'subscribe' AND channelPermissions.is_channel_available IS NULL
+        THEN 'channel_not_available'
+      -- otherwise, if channel is in a family without permissions for the user's org, then user can't subscribe it
+      WHEN adminUsers.is_admin_user IS NOT NULL
+        THEN NULL
+      -- otherwise, if channel does not have the "not_globally_subscribable" bit set, user can subscribe it
+      WHEN combination.role = 'subscribe' AND notGloballySubscribableChannels.is_not_globally_subscribable IS NULL
```
Comment (juliogonzalez) [inline schema/spacewalk/upgrade/susemanager-schema-4.1.10-to-susemanager-schema-4.1.11/001-fix_suseChannelUserRoleView.sql:36]: I basically run: `sed -i -e 's/notGlobSubsChannels/notGloballySubscribableChannels/g' *` Rest of the file is the same except the comments above (copyright updated, mention to oracle equivalent removed)

**Comment by juliogonzalez** [comment]
> Hmm, the last change dropped a different view as well.

That is right, but on paper that was already executed by the previous migration, and that's why I didn't included it.

But if you will feel safer having it, it won't hurt if I add it.

**Comment by juliogonzalez** [comment]
Ok, @mcalmer confirmed the schema_migration_tests_pgsql passes, so time to merge and then I will update the images.

**Pull request merged**